Extension/src/LanguageServer/client.ts

```diff
@@ -2324,6 +2325,7 @@ export class DefaultClient implements Client {
         this.languageClient.onNotification(ReportCodeAnalysisProcessedNotification, (e) => this.updateCodeAnalysisProcessed(e));
         this.languageClient.onNotification(ReportCodeAnalysisTotalNotification, (e) => this.updateCodeAnalysisTotal(e));
         this.languageClient.onNotification(DoxygenCommentGeneratedNotification, (e) => void this.insertDoxygenComment(e));
+        this.languageClient.onNotification(CanceledReferencesNotification, this.cancelReferences);
```

Comment:
Can you check if `workspaceReferences.referencesCanceled = true;` is set? That might need to be set, not sure.

Reply:
The call to `cancelReferences` will set `workspaceReferences.referencesCanceled = true` (if not already cancelling).  I tested the scenario, and the cancelation appears to work as expected.